custom/userAuth.js

```diff
@@ -50,7 +52,34 @@ getAuth().then(({email, key}) => {
 
   // seralize/deseralization methods for extracting user information from the
   // session cookie and adding it to the req.passport object
-  passport.serializeUser((user, done) => done(null, user))
+
+  // serialize is called when the user first authenticates
+  // Use this callback as an opportunity to check their email against the chive permissions list
+  // and attach an "authorized" flag to their PassportJS object
+  passport.serializeUser(async (user, done) => {
+    user.authorized = false;
+
+    // magic Google Apps Script; takes a "user" parameter in a GET request and returns a JSON object with an "authorized" key
+    const authUrl = "https://script.google.com/macros/s/AKfycby3zj7uw2ZHi6JUcCt8aaRoE3Z8lUo3uKojxGXXIgeCoJ9h78CvXGJasUl--t23JfxB/exec"
+
+    try {
+      // TODO: Technically there could be more than one email attached to a profile - should we send all of them?
+      var authResp = await axios.get(authUrl, {params: {user: user.emails[0].value}})
+
+      if (authResp.data.authorized) {
+        user.authorized = true;
+      }
+    } catch (e) {
+      console.log(e)
+    }
```

Comment:
I'm not familiar with Passport so I don't know if this is an issue, but the catch block here is falling through into line 77, so if there's an error you're still going to be calling `done` in the same way you would if there wasn't an error.

Reply:
That's intentional. We have to call `done` in order for passportjs to work correctly, but `user.authorized` will remain `false`.